content/docs/engine-dev/tools/tiaf/architectural-overview/runtime-frontend/_index.md

```diff
@@ -0,0 +1,335 @@
+
+##### Can file artifacts produced by jobs that do not necessarily correlate directly with jobs be consumed?
+
+Yes, in fact the Python instrumented test runner does this. All you need to do is implement an appropriate `PayloadMapProducer` to handle this as the `PayloadMapProducer` is provided with all of the completed jobs so it is up to you how you will then produce the appropriate payload map.
+
+##### The JobInfos contain the id and command string, yet the JobInfos themselves are template specialization, thus said ids and command strings are not interchangeable between different JobInfo types. Is this intentional?
+
+Yes. Although under the hood the `ProcessScheduler` understands the same ids and command strings, the JobInfos are not interchangeable nor convertible between specializations in order enforce the idea that a given job id or command string is only valid for that job type and not to be accidentally mixed up with another job type (e.g. if attempting to use the same job callback for different job types).
+
+### TestJobRunner
+
+The `TestJobRunner` is an intermediary base class for the `TestEnumerator` and `TestRunnerBase` that interacts with the `JobRunner` that it owns. It contains no public interface and only the shared state between these higher level classes. As such, there are no inputs, process or outputs for this class.
+
+### TestEnumerator
+
+The `TestEnumerator` is the base class derived from the TestJobRunner class template and partially specializes it for test frameworks that support the enumeration of tests from the command line. Once enumerated, the results may optionally be cached on the disk and re-read by future enumerations instead of invoking the test framework enumeration from the command line. Enumerated tests are stored in-memory in `TestEnumeration` data structures that contain the fixtures and tests that belong to those fixtures, as well as information about whether those tests/fixtures are enabled or disabled in the test framework.
+
+#### Inputs
+
+The `TestEnumerator` accepts the same inputs as the JobRunner.
+
+#### Process
+
+The `TestEnumerator` wraps around the JobRunner's call to run the jobs and injecting its `PayloadMapProducer` and `PayloadExtractor`. Optionally, it may either attempt to read enumerations from a cache file (specified by the client) and/or store the enumerations in said cache once enumerated.
+
+#### Outputs
+
+The `TestEnumerator` returns the same outputs as the JobRunner.
+
+#### FAQ
+
+##### You seem to have used the term `TestSuite` throughout the test and enumeration data structures instead of `TestFixture`. Why is this?
+
+This was a regrettable oversight from the early development days when the nomenclature borrowed from GTest's naming conventions. These will be renamed `TestFixture" in a future PR as we also have the concept of actual test suites and the current name conflict can lead to confusion.
```

Comment:
update: ` -> " mismatch for `TestFixture`

Reply:
Fixed 👍